### PR TITLE
Springboot bumps

### PR DESCRIPTION
**Pull request by matzew**
# Changes

- :broom: Update JDK version to 21 and go with suggested newer versions.

/kind <kind>

Fixes #

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

**Comment by matzew** [comment]
/assign @trisberg 
please review

/hold

File: templates/springboot/cloudevents/pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <parent>
     <groupId>org.springframework.boot</groupId>
     <artifactId>spring-boot-starter-parent</artifactId>
-    <version>3.1.5</version>
+    <version>3.2.1</version>
```
Comment (trisberg) [inline templates/springboot/cloudevents/pom.xml:7]: Just FYI, there is a brand new 3.2.2 release of Spring Boot now

File: templates/springboot/http/pom.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
   <parent>
     <groupId>org.springframework.boot</groupId>
     <artifactId>spring-boot-starter-parent</artifactId>
-    <version>3.1.5</version>
+    <version>3.2.1</version>
```
Comment (trisberg) [inline templates/springboot/http/pom.xml:7]: Just FYI, there is a brand new 3.2.2 release of Spring Boot now

File: templates/springboot/cloudevents/pom.xml (hunk 1 above)
Comment (matejvasek) [reply templates/springboot/cloudevents/pom.xml:7]: We might keep this to check if auto-updater changes this.

File: templates/springboot/http/pom.xml (hunk 2 above)
Comment (matejvasek) [reply templates/springboot/http/pom.xml:7]: We might keep this to check if auto-updater changes this.

**Pull request merged**